pkg/bootstrap/bootstrap.go

```diff
@@ -0,0 +1,181 @@
+// Copyright © 2022 sealos.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bootstrap
+
+import (
+	"context"
+	"fmt"
+
+	"golang.org/x/sync/errgroup"
+
+	v2 "github.com/labring/sealos/pkg/types/v1beta1"
+	"github.com/labring/sealos/pkg/utils/logger"
+)
+
+type Phase string
+
+const (
+	Preflight Phase = "preflight"
+	Init      Phase = "init"
+	Addon     Phase = "addon"
+)
+
+type Interface interface {
+	Preflight(hosts ...string) error
+	Init(hosts ...string) error
+	RegisterApplier(Phase, ...Applier) error
```

Comment:
Is this interface meaningful? Or change your name? It seems to be a little similar to registry.